toolkit/tools/pkg/imagecustomizerlib/partitionutils.go

```diff
@@ -352,6 +371,136 @@ func findPartition(mountIdType imagecustomizerapi.MountIdentifierType, mountId s
 	return partition, partitionIndex, nil
 }
 
+func findExtendedPartition(mountIdType imagecustomizerapi.ExtendedMountIdentifierType, mountId string,
+	partitions []diskutils.PartitionInfo, buildDir string,
+) (diskutils.PartitionInfo, int, error) {
+	matchedPartitionIndexes := []int(nil)
+	for i, partition := range partitions {
+		matches := false
+		switch mountIdType {
+		case imagecustomizerapi.ExtendedMountIdentifierTypeUuid:
+			matches = partition.Uuid == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypePartUuid:
+			matches = partition.PartUuid == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypePartLabel:
+			matches = partition.PartLabel == mountId
+		case imagecustomizerapi.ExtendedMountIdentifierTypeDev:
+			matches, err := findExtendedDevPartition(partition, partitions, buildDir)
```

Comment:
You should only do the conversion from /dev path to UUID a single time. That is, `findExtendedDevPartition` should be pulled outside of the loop.

And if you are feeling fancy, you could parse the value of `systemd.verity_root_data` and then feed it back into `findSourcePartitionHelper`.